src/core/tests/test_templatetags.py

```diff
@@ -28,3 +33,76 @@ def test_stateless_site_url_for_journal_path(self):
         url_name = 'journal_users' 
         url = fqdn.stateless_site_url(self.journal_two, url_name)
         self.assertEqual(url, f'http://{self.journal_two.press.domain}/{self.journal_two.code}/user/all/')
+
+
+@freeze_time("2025-03-26 12:00:00")
+class TestOffsetDateTag(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.user_default = helpers.create_user("default@example.com")
+        cls.user_tz = helpers.create_user(
+            "tz@example.com",
+            preferred_timezone="Europe/London",
+        )
+        cls.user_bad_tz = helpers.create_user(
+            "bad@example.com",
+            preferred_timezone="MemoryAlpha/Library",
+        )
+
+        cls.request_default = type("Request", (), {"user": cls.user_default})()
+        cls.request_tz = type("Request", (), {"user": cls.user_tz})()
+        cls.request_bad_tz = type("Request", (), {"user": cls.user_bad_tz})()
+
+    def test_offset_date_with_default_timezone_datetime(self):
+        """
+        Test offset_date tag with default timezone and datetime-local input.
+        """
+        context = {"request": self.request_default}
+        result = dates.offset_date(
+            context,
+            days=2,
+            input_type="datetime-local",
+        )
+        expected = "2025-03-28T12:00"
+        self.assertEqual(result, expected)
+
+    def test_offset_date_with_preferred_timezone_datetime(self):
+        """
+        Test offset_date tag with a valid preferred timezone and datetime-local input.
+        """
+        context = {"request": self.request_tz}
+        # 12:00 UTC is 12:00 GMT in March (not in daylight saving yet)
+        result = dates.offset_date(
+            context,
+            days=1,
+            input_type="datetime-local",
+        )
+        expected = "2025-03-27T12:00"
```

Comment:
given that we are freezing time with a timezone-unaware string, I believe we could have unintended consequences for the translation between the host timezone and the timezone selected for the test user `"Europe/London"`. Namely daylight savings - If I'm reading all this correctly the result will be one hour off the expected for 6 months every year

Reply:
Ah, since we are freezing the date, that part becomes a non-issue, but I think we still have an issue if the host system date is not in UTC when running the test